### ISSUE TITLE
Reduce requests for widgets and operators

### ISSUE THREAD

**Issue by DerJayDee** [Wirecloud/wirecloud#213]
For every widget the `WirecloudAPIBootstrap.js` file is requested. For every operator the `WirecloudOperatorAPI.js` file is requested. Both files are not compressed although `DEBUG = False` is set in the `settings.py`.
It would be great if those files would also get compressed by the `django-compressor`.

It would be even greater when those files would get requested only once for the whole dashboard.

And to top it all off: It would be perfect if there is a possibility of sharing JS-Scripte in between widgets / operators. For example jQuery is probably used in most widgets.

So I have a dream that one day a wirecloud admin is able to specify a folder on the wirecloud host system that contains JavaScript files that get compressed and are shared among all widgets and operators.

**Comment by aarranz** [comment]
Thanks for reporting this; we are going to work on it as soon as possible.

We are going to work on creating a compressed file from `WirecloudAPIBootstrap.js`, `WirecloudWidgetAPI.js` and `WirecloudAPICommon.js` for widgets and a compressed file from `WirecloudAPIBootstrap.js`, `WirecloudOperatorAPI.js` and `WirecloudAPICommon.js` for operators.

> It would be even greater when those files would get requested only once for the whole dashboard.

Well, those files are currently requested only one time as much, at least if you follow the installation instructions, as they are configured to be cached for a year. I'm guessing you means to execute this file only one time... but I'm afraid that this cannot be accomplished using current browsers (i.e. without support for shadow DOM or anything similar) and with the current WireCloud architecture. Anyway, those files are currently taking ~20ms to load (per widget/operator) once cached, we'll see how much we are able to reduce this time :).

> And to top it all off: It would be perfect if there is a possibility of sharing JS-Scripte in between widgets / operators. For example jQuery is probably used in most widgets.

This also seems a great idea 👍, but probably this is better tracked by creating a new ticket.
